docs/controller.js

```diff
@@ -370,4 +371,14 @@ function initPage() {
             drawCtx.clearRect(0, 0, width, height);
         }
     });
+    function resizeHandler(){
+        var canvas = document.querySelector('#viewport')
+        canvas.style.height = innerHeight - 60 + 'px';
```

Comment:
Maybe make these into `calc` expressions?

Do canvases even respond properly to styles or do they stretch the content? You might need to apply the height and width to the canvas itself so that you don't get scaling artifacts on high dpi devices.